lib/daru/maths/statistics/vector.rb

```diff
@@ -377,6 +377,40 @@ def sample_without_replacement(sample=1)
           end
         end
 
+        # The percent_change method computes the percent change over
+        # the given number of periods.
+        #
+        # @param [Integer] periods (1) number of nils to insert at the beginning.
+        #
+        # @example
+        # 
+        #   vector = Daru::Vector.new([4,6,6,8,10],index: ['a','f','t','i','k'])
+        #   vector.percent_change
+        #   #=>
+        #   #   <Daru::Vector:28713060 @name = nil @size: 5 >
+        #   #              nil
+        #   #   a	
+        #   #   f	   0.5
+        #   #   t	   0.0
+        #   #   i	   0.3333333333333333
+        #   #   k          0.25
+        def percent_change periods = 1
+          type == :numeric or raise TypeError, "Vector must be numeric"
+          value = self.only_valid
```

Comment:
This code won't work for test cases like this : 
`vector = Daru::Vector.new([2,4,5,nil,6,7,nil])`
Actually, counter `i` must run through only numbers excluding the nils. So working with `only_valid` will be the best in this case. I initially tried to do this with deleting nils from `@data` but that is obviously not the way out.